library/std/src/f64.rs

```diff
@@ -936,8 +931,8 @@ impl f64 {
     // of expected NaN).
     #[rustc_allow_incoherent_impl]
     fn log_wrapper<F: Fn(f64) -> f64>(self, log_fn: F) -> f64 {
-        if !cfg!(any(target_os = "solaris", target_os = "illumos")) {
-            log_fn(self)
+        if let Some(result) = crate::sys::log_wrapper(self) {
```

Comment:
i think you flipped the logic here.

Reply:
oh nice catch, that actually makes this a lot easier to abstract :smile: